api/src/transactions.rs

```diff
@@ -1232,7 +1233,22 @@ impl TransactionsApi {
                 let mut user_transactions = Vec::new();
                 for transaction in transactions.into_iter() {
                     match transaction {
-                        Transaction::UserTransaction(user_txn) => user_transactions.push(*user_txn),
+                        Transaction::UserTransaction(user_txn) => {
+                            let mut txn = *user_txn;
+                            match &vm_status {
+                                VMStatus::Error {
```

Comment:
You can use `!vm_status.is_success()` to catch `MoveAbort`s as well

Reply:
Let's try to use whatever @georgemitenkov has refactored it to be to be consistent.